modules/tinymce/src/core/main/ts/mode/Readonly.ts

```diff
@@ -26,7 +26,8 @@ const toggleReadOnly = (editor: Editor, state: boolean): void => {
 const isReadOnly = (editor: Editor): boolean => editor.readonly;
 
 const registerReadOnlyInputBlockers = (editor: Editor): void => {
-  editor.on('beforeinput paste cut dragend dragover draggesture dragdrop drop drag', (e) => {
+  // Block all input events in readonly mode
+  editor.on('beforeinput input paste cut dragend dragover draggesture dragdrop drop drag', (e) => {
```

Comment:
The `input` event is non cancellable see: https://w3c.github.io/uievents/#event-type-input `beforeinput` is on most platforms but not Android.